test/tuple.cpp

```diff
@@ -360,9 +360,53 @@ TEST(CampTuple, StructuredBindings)
 {
   auto t = camp::make_tuple(3, 9.9);
   auto [a, b] = t;
+  static_assert(std::is_same_v< decltype(a), int > );
+  static_assert(std::is_same_v< decltype(b), double > );
```

Comment:
Unfortunately, we currently still have to support c++14, so these need to downgrade. Otherwise this all looks great.